cmd/swarm/config.go

```diff
@@ -190,50 +190,44 @@ func flagsOverride(currentConfig *bzzapi.Config, ctx *cli.Context) *bzzapi.Confi
 			currentConfig.NetworkID = id
 		}
 	}
-
 	if ctx.GlobalIsSet(utils.DataDirFlag.Name) {
 		if datadir := ctx.GlobalString(utils.DataDirFlag.Name); datadir != "" {
 			currentConfig.Path = expandPath(datadir)
 		}
 	}
-
 	bzzport := ctx.GlobalString(SwarmPortFlag.Name)
 	if len(bzzport) > 0 {
 		currentConfig.Port = bzzport
 	}
-
 	if bzzaddr := ctx.GlobalString(SwarmListenAddrFlag.Name); bzzaddr != "" {
 		currentConfig.ListenAddr = bzzaddr
 	}
-
 	if ctx.GlobalIsSet(SwarmSwapEnabledFlag.Name) {
 		currentConfig.SwapEnabled = true
 	}
-
+	if swapBackendURL := ctx.GlobalString(SwarmSwapBackendURLFlag.Name); swapBackendURL != "" {
+		currentConfig.SwapBackendURL = swapBackendURL
+	}
+	if paymentThreshold := ctx.GlobalUint64(SwarmSwapPaymentThresholdFlag.Name); paymentThreshold != 0 {
+		currentConfig.SwapPaymentThreshold = paymentThreshold
+	}
+	if disconnectThreshold := ctx.GlobalUint64(SwarmSwapDisconnectThresholdFlag.Name); disconnectThreshold != 0 {
+		currentConfig.SwapDisconnectThreshold = disconnectThreshold
+	}
 	if ctx.GlobalIsSet(SwarmSyncDisabledFlag.Name) {
 		currentConfig.SyncEnabled = false
 	}
-
 	if d := ctx.GlobalDuration(SwarmSyncUpdateDelay.Name); d > 0 {
 		currentConfig.SyncUpdateDelay = d
 	}
-
 	// any value including 0 is acceptable
 	currentConfig.MaxStreamPeerServers = ctx.GlobalInt(SwarmMaxStreamPeerServersFlag.Name)
-
 	if ctx.GlobalIsSet(SwarmLightNodeEnabled.Name) {
 		currentConfig.LightNodeEnabled = true
 	}
-
 	if ctx.GlobalIsSet(SwarmDeliverySkipCheckFlag.Name) {
 		currentConfig.DeliverySkipCheck = true
 	}
-
-	currentConfig.SwapBackendURL = ctx.GlobalString(SwarmSwapBackendURLFlag.Name)
-	if currentConfig.SwapEnabled && currentConfig.SwapBackendURL == "" {
```

Comment:
Integrity is verified by SWAP

Reply:
what do you mean by this?

Reply:
Verification is now in the function `NewSWAP`. It was before already in NewSwarm (it was in two places). 

@holisticode requested me to add a check whether the `paymentThreshold` is lower than the `disconnectThreshold`. I was thinking about where to do this, and didn't find it fitting in either the file `cmd/swarm/config.go` and `swarm.go`. Instead, I replaced all tests which were previously tested with regards to SWAP variables + the test on the thresholds to a new function `NewSWAP` in `swap.go`

Reply:
Yes I think that is a good solution